lib/usb/Makefile

```diff
@@ -1,32 +1,59 @@
-USB_SRCS := \
-	$(NXDK_DIR)/lib/usb/host/ohci-hcd.c \
-	$(NXDK_DIR)/lib/usb/core/message.c \
-	$(NXDK_DIR)/lib/usb/core/hcd.c \
-	$(NXDK_DIR)/lib/usb/core/hcd-pci.c \
-	$(NXDK_DIR)/lib/usb/core/hub.c \
-	$(NXDK_DIR)/lib/usb/core/usb.c \
-	$(NXDK_DIR)/lib/usb/core/config.c \
-	$(NXDK_DIR)/lib/usb/core/urb.c \
-	$(NXDK_DIR)/lib/usb/core/buffer_simple.c \
-	$(NXDK_DIR)/lib/usb/core/usb-debug.c \
-	$(NXDK_DIR)/lib/usb/sys/BootUSB.c \
-	$(NXDK_DIR)/lib/usb/sys/linuxwrapper.c \
-	$(NXDK_DIR)/lib/usb/sys/xpad.c \
-	$(NXDK_DIR)/lib/usb/sys/xremote.c \
-	$(NXDK_DIR)/lib/usb/sys/usbkey.c \
-	$(NXDK_DIR)/lib/usb/sys/usbmouse.c \
-	$(NXDK_DIR)/lib/usb/misc/misc.c \
-	$(NXDK_DIR)/lib/usb/misc/pci.c \
-	$(NXDK_DIR)/lib/usb/misc/malloc.c
-
-USB_OBJS = $(addsuffix .obj, $(basename $(USB_SRCS)))
-
-$(NXDK_DIR)/lib/libnxdk_usb.lib: $(USB_OBJS)
-
-main.exe: $(NXDK_DIR)/lib/libnxdk_usb.lib
-
-CLEANRULES += clean-usb
-
-.PHONY: clean-usb
-clean-usb:
-	$(VE)rm -f $(USB_OBJS) $(NXDK_DIR)/lib/libnxdk_usb.lib
+USBH_DIR=$(NXDK_DIR)/lib/usb/ohci_usbhostlib
+
+#USB Host core
+USB_CORE_SRCS := \
+	$(USBH_DIR)/src_core/hub.c \
+	$(USBH_DIR)/src_core/mem_alloc.c \
+	$(USBH_DIR)/src_core/ohci.c \
+	$(USBH_DIR)/src_core/support.c \
+	$(USBH_DIR)/src_core/usb_core.c
+
+#USB Human Input Device class driver
+USB_HID_SRCS := \
+	$(USBH_DIR)/src_hid/hid_core.c \
+	$(USBH_DIR)/src_hid/hid_driver.c
+
+#USB Communication Device class driver
+USB_CDC_SRCS := \
+	$(USBH_DIR)/src_cdc/cdc_core.c \
+	$(USBH_DIR)/src_cdc/cdc_driver.c \
+	$(USBH_DIR)/src_cdc/cdc_parser.c
+	
+#USB Audio class driver
+USB_UAC_SRCS := \
+	$(USBH_DIR)/src_uac/uac_core.c \
+	$(USBH_DIR)/src_uac/uac_driver.c \
+	$(USBH_DIR)/src_uac/uac_parser.c
+
+# USB Video class driver
+USB_UVC_SRCS := \
+	$(USBH_DIR)/src_uvc/uvc_core.c \
+	$(USBH_DIR)/src_uvc/uvc_driver.c \
+	$(USBH_DIR)/src_uvc/uvc_parser.c
+
+# Mass Storage Class driver
+# FIXME: This driver is currently hardcoded so that it requires FATFS
+# This requirement should be removed and abstracted out.
+# USB_MSC_SRCS := \
+#	$(USBH_DIR)/src_msc/msc_driver.c \
+#	$(USBH_DIR)/src_msc/msc_xfer.c
+
+USBH_OBJS = $(addsuffix .obj, $(basename $(USB_CORE_SRCS)))
+USBH_OBJS += $(addsuffix .obj, $(basename $(USB_HID_SRCS)))
+USBH_OBJS += $(addsuffix .obj, $(basename $(USB_CDC_SRCS)))
+USBH_OBJS += $(addsuffix .obj, $(basename $(USB_UAC_SRCS)))
+USBH_OBJS += $(addsuffix .obj, $(basename $(USB_UVC_SRCS)))
+#USBH_OBJS += $(addsuffix .obj, $(basename $(USB_MSC_SRCS)))
+
+$(NXDK_DIR)/lib/libnxdk_usb.lib: $(USBH_OBJS)
+
+NXDK_CFLAGS += -I$(USBH_DIR)/inc
+NXDK_CXXFLAGS += -I$(USBH_DIR)/inc
+
+main.exe: $(NXDK_DIR)/lib/libnxdk_usb.lib
+
+CLEANRULES += clean-usb
+
+.PHONY: clean-usb
+clean-usb:
+	$(VE)rm -f $(USBH_OBJS) $(NXDK_DIR)/lib/libnxdk_usb.lib

```

Comment:
I think this Makefile should be part of the USB submodule, unless that submodule remains platform unspecific (which, to be honest, sounds like a good idea to me; but only if we can attract other stakeholders / convince upstream to submodule it)

Other maintainers might have different thoughts on this matter though.